pkg/vault-mon/prometheus.go

```diff
@@ -83,14 +83,30 @@ func PromWatchCerts(pkimon *PKIMon, interval time.Duration) {
 
 						crl_expiry.WithLabelValues(pkiname, issuer).Set(float64(crl.NextUpdate.Sub(now).Seconds()))
 						crl_nextupdate.WithLabelValues(pkiname, issuer).Set(float64(crl.NextUpdate.Unix()))
-						crl_length.WithLabelValues(pkiname, issuer).Set(float64(len(crl.RevokedCertificates)))
+						crl_length.WithLabelValues(pkiname, issuer).Set(float64(len(crl.RevokedCertificateEntries)))
 						crl_byte_size.WithLabelValues(pkiname, issuer).Set(float64(pki.crlRawSize))
 
 						slog.Debug("Updated CRL metrics", "pki", pkiname, "issuer", issuer, "next_update", crl.NextUpdate)
 
 						// gather revoked certs from the CRL so we can exclude their metrics later
-						for _, revokedCert := range crl.RevokedCertificates {
+						for _, revokedCert := range crl.RevokedCertificateEntries {
+
+							// loadCerts() also excludes revoked certs from the cert map
+							// but this goes an extra step and deletes certificate metrics on every Prometheus refresh interval instead
+							// must not compare to GetCerts() map but just from each CRL load
 							revokedCerts[revokedCert.SerialNumber.String()] = struct{}{}
+
+							// Only know serial number natively from revokedCert object
+							labels := prometheus.Labels{
+								"serial": strings.ReplaceAll(fmt.Sprintf("% x", revokedCert.SerialNumber.Bytes()), " ", "-"),
```

Comment:
it allow us to remove all metric link to this serial ?

Reply:
yes exactly! any time series with the `serial` label of a revoked certificate will be deleted